### PR TITLE
Document the service rebinding behavior

### PR DESCRIPTION
**Pull request by Srivaralakshmi**
[RHDEVDOCS-4257](https://issues.redhat.com/browse/RHDEVDOCS-4257): Document a note on the service rebinding behavior

- **Aligned team**: Dev Tools
- **OCP version for cherry-picking**: enterprise-`4.10`, enterprise-`4.11` and later
- **JIRA issues**: [RHDEVDOCS-4257](https://issues.redhat.com/browse/RHDEVDOCS-4257)
- **Preview pages**: [Download to see the preview in your browser](https://drive.google.com/file/d/1t3ZSCZXhyBdN80rgN46CbRAIXwmIY_A-/view?usp=sharing)
- **SME Review**: Completed by @baijum, @dperaza4dustbit
- **QE review**: Completed by @pmacik 
- **Peer-review**: Completed by @rolfedh

### CODE REVIEW

File: modules/sbo-advanced-binding-options.adoc [hunk 1]
```diff
@@ -180,4 +180,7 @@ If you define the following pairs of fields, {servicebinding-title} refuses the
 
 * The `name` and `labelSelector` fields in the `binding.operators.coreos.com/v1alpha1` API.
 * The `name` and `selector` fields in the `servicebinding.io` API (Spec API).
-====
\ No newline at end of file
+====
+
+.Understanding the rebinding behavior
+Consider a case where after a successful binding, you are using the `name` field to identify a workload. In such a case, if you delete an existing workload and recreate it, the projected binding data are lost. The `ServiceBinding` reconciler does not rebind the workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.
```
Comment (pmacik) [inline modules/sbo-advanced-binding-options.adoc:186]:
```suggestion
Consider a case where after a successful binding, you are using the `name` field to identify a workload. In such a case, if you delete an existing workload and recreate it, the `ServiceBinding` reconciler does not rebind the workload and so the binding data are not projected to the recreated workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.
```

Comment (Srivaralakshmi) [reply modules/sbo-advanced-binding-options.adoc:186]: Thanks, @pmacik for the suggestion. I have implemented it but by tweaking it a little for grammar and language.

**Comment by baijum** [comment]
Change to "rebinding" in the title of this PR.

**Review by dperaza4dustbit — APPROVED** [review]
lgtm

**Review by baijum — APPROVED** [review]
LGTM

**Comment by Srivaralakshmi** [comment]
Hi, @baijum and @pmacik: Thanks for your comments. I have implemented them. PTAL at the revised PR and provide your feedback.

Hi, @dperaza4dustbit: PTAL at the PR and provide your feedback. Thanks!

File: modules/sbo-advanced-binding-options.adoc [hunk 2]
```diff
@@ -180,4 +180,7 @@ If you define the following pairs of fields, {servicebinding-title} refuses the
 
 * The `name` and `labelSelector` fields in the `binding.operators.coreos.com/v1alpha1` API.
 * The `name` and `selector` fields in the `servicebinding.io` API (Spec API).
-====
\ No newline at end of file
+====
+
+.Understanding the rebinding behavior
+Consider a case where after a successful binding you use the `name` field to identify a workload. In such a case, if you delete an existing workload and recreate it, the `ServiceBinding` reconciler does not rebind the workload and therefore the Operator does not project the binding data to the workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.
```
Comment (rolfedh) [inline modules/sbo-advanced-binding-options.adoc:186]:
```suggestion
Consider a case where, after a successful binding, you use the `name` field to identify a workload. If you delete and recreate that workload, the `ServiceBinding` reconciler does not rebind the workload, and the Operator cannot project the binding data to the workload. However, if you use the `labelSelector` field to identify a workload, the `ServiceBinding` reconciler rebinds the workload, and the Operator projects the binding data.
```

**Review by rolfedh — APPROVED** [review]
Suggestions for minimalism. You're welcome to apply what seems right to you.

File: modules/sbo-advanced-binding-options.adoc (hunk 2 above)
Comment (Srivaralakshmi) [reply modules/sbo-advanced-binding-options.adoc]: Done

**Comment by Srivaralakshmi** [comment]
Thanks, @rolfedh for your suggestions. I have addressed them. PTAL at the revised PR and provide your approval.

**Review by rolfedh — APPROVED** [review]

**Pull request merged**